### ISSUE TITLE
Getting metadata or deleting empty folder will result 404

### ISSUE THREAD

**Issue by RezaRahmati** [angular/angularfire#3267]
### Version info

Angular: 14.1.3

Firebase: 9.9.3

AngularFire: 7.4.1

**Other (e.g. Ionic/Cordova, Node, browser, operating system):**

### How to reproduce these conditions

1. Create a folder in cloud storage
2. then run below code
```
import { AngularFireStorage } from '@angular/fire/compat/storage';
       ...
	constructor(private storage: AngularFireStorage) {}
       ....
		const ref = this.storage.ref(pathToFolder);
                ref.getMetadata().toPromise(); // this will result a GET request with 404 as response
                ref.delete().toPromise(); // this will result a DELETE request with 404 as response
                
```

**Failing test unit, Stackblitz demonstrating the problem**

**Steps to set up and reproduce**

**Sample data and security rules**

### Debug output

** Errors in the JavaScript console **

** Output from `firebase.database().enableLogging(true);` **

** Screenshots **
![image](https://user-images.githubusercontent.com/10086092/187002212-b176bd89-2411-47e8-98af-4b76fc57429d.png)

**Comment by jnizet** [comment]
Firebase Cloud STorage is basically the same as Google Cloud Storage, and there is no such thing as a directory in Google Cloud Storage. See https://cloud.google.com/storage/docs/objects and https://cloud.google.com/storage/docs/naming-objects for details.

**Comment by RezaRahmati** [comment]
@jnizet Yes I understand that, the issue is I am bale to get the ref but not able to do the operation. I tried adding trailing slash to the path, but still same

I am able to do the same on backend using the below code, and ended up using the api to delete the folder
```
    async deleteFolder(bucketName: string, folder: string): Promise<void> {
        const bucket = this.storage.bucket(bucketName);
        const [files] = await bucket.getFiles({ prefix: `${folder}/` });
        await Promise.allSettled(files.map(file => file.delete({ ignoreNotFound: true })))
    }
```

**Comment by jnizet** [comment]
That code deletes all the objects under a prefix. It's not the same as trying to delete a folder, because folders don't exist.

**Comment by RezaRahmati** [comment]
@jnizet would you do me a favor and share a working code snippet that deletes a folder on cloud storage successfully?

**Comment by jnizet** [comment]
No, because there is no such thing as a folder on cloud storage. You can't delete something that doesn't exist.

**Comment by RezaRahmati** [comment]
my goal is to delete this

![image](https://user-images.githubusercontent.com/10086092/188498745-48437bd9-da0f-4e53-bcf8-cab9642ca0b8.png)

**Comment by armando-navarro** [comment]
Cloud Storage has no real folders: an empty 'folder' is not a stored object, so calling `getMetadata()` or `delete()` on that path returning 404 is expected behavior rather than an AngularFire bug.

To remove the items under a prefix you have to delete the underlying objects, which is what your `getFiles({ prefix })` workaround does. If you run into a case on a current version where a real stored object 404s unexpectedly, please open a new issue with a repro.
